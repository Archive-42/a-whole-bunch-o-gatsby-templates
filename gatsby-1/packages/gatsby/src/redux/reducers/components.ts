import normalize from "./normalize-path"
import { IGatsbyState, ActionsUnion } from "../types"

let programStatus = `BOOTSTRAPPING`

// TODO: replace usages of this reducer with queries.trackedComponents
//  It is here merely for compatibility.
export const componentsReducer = (
  state: IGatsbyState["components"] = new Map(),
  action: ActionsUnion
): IGatsbyState["components"] => {
  switch (action.type) {
    case `DELETE_CACHE`:
      return new Map()
    case `SET_PROGRAM_STATUS`:
      programStatus = action.payload
      return state
    case `CREATE_PAGE`: {
      action.payload.componentPath = normalize(action.payload.component)
      // Create XState service.
      let component = state.get(action.payload.componentPath)
      if (!component) {
        component = {
          componentPath: action.payload.componentPath,
          componentChunkName: action.payload.componentChunkName,
          query: ``,
          pages: new Set(),
          isInBootstrap: true,
        }
      }
      component.pages.add(action.payload.path)
      component.isInBootstrap = programStatus === `BOOTSTRAPPING`
      state.set(action.payload.componentPath, component)
      return state
    }
    case `QUERY_EXTRACTED`: {
      action.payload.componentPath = normalize(action.payload.componentPath)
      const component = state.get(action.payload.componentPath)!
      component.query = action.payload.query
      state.set(action.payload.componentPath, component)
      return state
    }
    case `REMOVE_STATIC_QUERIES_BY_TEMPLATE`: {
      action.payload.componentPath = normalize(action.payload.componentPath)
      state.delete(action.payload.componentPath)
      return state
    }
    case `DELETE_PAGE`: {
      const component = state.get(normalize(action.payload.component))!
      component.pages.delete(action.payload.path)
      return state
    }
  }

  return state
}
