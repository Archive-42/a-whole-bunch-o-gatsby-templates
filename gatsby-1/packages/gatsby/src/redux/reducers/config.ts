import { IGatsbyConfig, ISetSiteConfig } from "../types"

export const configReducer = (
  state: IGatsbyConfig = {},
  action: ISetSiteConfig
): IGatsbyConfig => {
  switch (action.type) {
    case `SET_SITE_CONFIG`: {
      return action.payload
    }
    default:
      return state
  }
}
