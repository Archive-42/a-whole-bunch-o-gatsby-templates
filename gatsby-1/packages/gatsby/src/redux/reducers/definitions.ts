import { ActionsUnion, IGatsbyState } from "../types"

export const definitionsReducer = (
  state: IGatsbyState["definitions"] = new Map(),
  action: ActionsUnion
): IGatsbyState["definitions"] => {
  switch (action.type) {
    case `SET_GRAPHQL_DEFINITIONS`:
      return action.payload
    default:
      return state
  }
}
