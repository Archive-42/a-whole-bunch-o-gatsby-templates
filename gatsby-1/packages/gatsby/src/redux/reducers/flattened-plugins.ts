import { IGatsbyState, ActionsUnion } from "../types"

export const flattenedPluginsReducer = (
  state: IGatsbyState["flattenedPlugins"] = [],
  action: ActionsUnion
): IGatsbyState["flattenedPlugins"] => {
  switch (action.type) {
    case `SET_SITE_FLATTENED_PLUGINS`:
      return [...action.payload]

    default:
      return state
  }
}
