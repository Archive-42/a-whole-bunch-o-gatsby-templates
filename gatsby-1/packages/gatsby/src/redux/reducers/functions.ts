import { IGatsbyState, ISetSiteFunctions } from "../types"

export const functionsReducer = (
  state: IGatsbyState["functions"] = [],
  action: ISetSiteFunctions
): IGatsbyState["functions"] => {
  switch (action.type) {
    case `SET_SITE_FUNCTIONS`:
      return action.payload

    default:
      return state
  }
}
