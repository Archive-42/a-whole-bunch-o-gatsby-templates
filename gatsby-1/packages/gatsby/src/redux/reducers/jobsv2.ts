import {
  ICreateJobV2Action,
  IRemoveStaleJobV2Action,
  IEndJobV2Action,
  IGatsbyState,
  IGatsbyIncompleteJobV2,
  IGatsbyCompleteJobV2,
  IDeleteCacheAction,
} from "../types"

const initialState = (): IGatsbyState["jobsV2"] => {
  return {
    incomplete: new Map(),
    complete: new Map(),
  }
}

export const jobsV2Reducer = (
  state = initialState(),
  action:
    | ICreateJobV2Action
    | IRemoveStaleJobV2Action
    | IEndJobV2Action
    | IDeleteCacheAction
): IGatsbyState["jobsV2"] => {
  switch (action.type) {
    case `DELETE_CACHE`:
      return (action as IDeleteCacheAction).cacheIsCorrupt
        ? initialState()
        : state

    case `CREATE_JOB_V2`: {
      const { job } = action.payload

      state.incomplete.set(job.contentDigest, {
        job,
      } as IGatsbyIncompleteJobV2)

      return state
    }

    case `END_JOB_V2`: {
      const { jobContentDigest, result } = action.payload
      const { job } = state.incomplete.get(
        jobContentDigest
      ) as IGatsbyIncompleteJobV2

      if (!job) {
        throw new Error(
          `If you encounter this error, it's probably a Gatsby internal bug. Please open an issue reporting us this.`
        )
      }

      state.incomplete.delete(job.contentDigest)

      // inputPaths is used to make sure the job is not stale
      state.complete.set(job.contentDigest, {
        result,
        inputPaths: job.inputPaths,
      } as IGatsbyCompleteJobV2)

      return state
    }

    case `REMOVE_STALE_JOB_V2`: {
      const { contentDigest } = action.payload
      state.incomplete.delete(contentDigest)
      state.complete.delete(contentDigest)

      return state
    }
  }

  return state
}
