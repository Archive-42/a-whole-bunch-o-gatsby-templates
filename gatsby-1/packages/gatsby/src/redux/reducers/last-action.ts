import { IGatsbyState, ActionsUnion } from "../types"

export const lastActionReducer = (
  _state: unknown,
  action: ActionsUnion
): IGatsbyState["lastAction"] => action
