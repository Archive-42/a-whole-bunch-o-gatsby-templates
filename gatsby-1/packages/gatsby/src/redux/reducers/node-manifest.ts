import reporter from "gatsby-cli/lib/reporter"

import {
  IGatsbyState,
  ICreateNodeManifest,
  IDeleteNodeManifests,
} from "../types"

export const nodeManifestReducer = (
  state: IGatsbyState["nodeManifests"] = [],
  action: ICreateNodeManifest | IDeleteNodeManifests
): IGatsbyState["nodeManifests"] => {
  switch (action.type) {
    case `CREATE_NODE_MANIFEST`: {
      const { manifestId, pluginName, node } = action.payload

      if (typeof manifestId !== `string`) {
        reporter.warn(
          `Plugin ${pluginName} called unstable_createNodeManifest with a manifestId that isn't a string.`
        )

        return state
      }

      if (!node?.id) {
        reporter.warn(
          `Plugin ${pluginName} called unstable_createNodeManifest but didn't provide a node.`
        )

        return state
      }

      state.push({
        ...action.payload,
        node: {
          id: node.id,
        },
      })

      return state
    }

    case `DELETE_NODE_MANIFESTS`: {
      state = []

      return state
    }

    default:
      return state
  }
}
