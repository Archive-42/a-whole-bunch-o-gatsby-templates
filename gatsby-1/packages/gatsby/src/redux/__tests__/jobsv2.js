const jobsManager = require(`../../utils/jobs/manager`)
jest.spyOn(jobsManager, `enqueueJob`)
jest.spyOn(jobsManager, `removeInProgressJob`)
jest.mock(`uuid/v4`, () => () => `1234`)

import { jobsV2Reducer as jobsReducer } from "../reducers/jobsv2"

describe(`Job v2 actions/reducer`, () => {
  const plugin = {
    name: `test-plugin`,
    version: `1.0.0`,
    resolve: `/node_modules/test-plugin`,
  }
  const createGlobalState = defaultState => {
    return {
      program: {
        directory: __dirname,
      },
      jobsV2: jobsReducer(undefined, {}),
      ...defaultState,
    }
  }

  beforeEach(() => {
    jobsManager.enqueueJob.mockClear()
    jobsManager.enqueueJob.mockImplementation(() => Promise.resolve(`myresult`))
  })

  const createDispatcher = globalState => {
    const dispatch = realAction => jobsReducer(getState().jobsV2, realAction)
    const getState = () => globalState

    return action => action(dispatch, getState)
  }

  const getIsolatedActions = () => {
    let allActions
    jest.isolateModules(() => {
      const { actions, internalActions } = require(`../actions`)
      allActions = { actions, internalActions }
    })

    return allActions
  }

  it(`should enqueueJob`, async () => {
    const { actions } = getIsolatedActions()
    const globalState = createGlobalState()
    const dispatch = createDispatcher(globalState)

    const promise = dispatch(
      actions.createJobV2(
        {
          name: `TEST_JOB`,
          inputPaths: [],
          outputDir: `/public/static`,
          args: {},
        },
        plugin
      )
    )

    expect(globalState.jobsV2).toMatchSnapshot()

    expect(globalState.jobsV2.complete.size).toBe(0)
    expect(globalState.jobsV2.incomplete.size).toBe(1)

    await promise

    const values = Array.from(globalState.jobsV2.complete.values())
    expect(globalState.jobsV2.complete.size).toBe(1)
    expect(values[0]).toEqual({
      result: `myresult`,
      inputPaths: [],
    })
    expect(jobsManager.removeInProgressJob).toHaveBeenCalledTimes(1)
    expect(jobsManager.enqueueJob).toMatchSnapshot()
  })

  it(`should return the result when job already ran`, async () => {
    const { actions } = getIsolatedActions()
    const globalState = createGlobalState()
    const dispatch = createDispatcher(globalState)

    const job = {
      name: `TEST_JOB`,
      inputPaths: [],
      outputDir: `/public/static`,
      args: {},
    }

    await expect(dispatch(actions.createJobV2(job, plugin))).resolves.toBe(
      `myresult`
    )
    await expect(dispatch(actions.createJobV2(job, plugin))).resolves.toBe(
      `myresult`
    )
    expect(jobsManager.enqueueJob).toHaveBeenCalledTimes(1)
  })

  it(`should remove a stale job`, async () => {
    const { internalActions } = getIsolatedActions()
    const { jobsV2 } = createGlobalState()
    jobsV2.complete.set(`1234`, {})
    jobsV2.incomplete.set(`12345`, {})

    jobsReducer(jobsV2, internalActions.removeStaleJob(`1234`))
    expect(jobsV2.complete.size).toBe(0)
    expect(jobsV2.incomplete.size).toBe(1)

    jobsReducer(jobsV2, internalActions.removeStaleJob(`12345`))
    expect(jobsV2.complete.size).toBe(0)
    expect(jobsV2.incomplete.size).toBe(0)
  })
})
