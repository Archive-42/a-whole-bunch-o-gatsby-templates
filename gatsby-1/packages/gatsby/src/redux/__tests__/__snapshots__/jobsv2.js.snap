// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job v2 actions/reducer should enqueueJob 1`] = `
Object {
  "complete": Map {},
  "incomplete": Map {
    "aa9932d515707737e953173cd9c77306" => Object {
      "job": Object {
        "args": Object {},
        "contentDigest": "aa9932d515707737e953173cd9c77306",
        "id": "1234",
        "inputPaths": Array [],
        "name": "TEST_JOB",
        "outputDir": "/public/static",
        "plugin": Object {
          "isLocal": false,
          "name": "test-plugin",
          "resolve": "/node_modules/test-plugin",
          "version": "1.0.0",
        },
      },
    },
  },
}
`;

exports[`Job v2 actions/reducer should enqueueJob 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {},
        "contentDigest": "aa9932d515707737e953173cd9c77306",
        "id": "1234",
        "inputPaths": Array [],
        "name": "TEST_JOB",
        "outputDir": "/public/static",
        "plugin": Object {
          "isLocal": false,
          "name": "test-plugin",
          "resolve": "/node_modules/test-plugin",
          "version": "1.0.0",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
